### PR TITLE
Initial draft

### PR DESCRIPTION
**Pull request by danvangeest**
Strongly based on lamps-dilithium-certificates and lamps-cms-sphincs-plus

### CODE REVIEW

File: draft-gazdag-x509-slhdsa.md [hunk 1]
```diff
@@ -43,40 +54,434 @@ author:
...
+
+SLH-DSA is a hash-based signature scheme which consists of a few time signature construction, namely Forest of Random Subsets (FORS) and a hypertree.  FORS signs a message with a private key.  The corresponding FORS public keys are the leaves in k binary trees.  The roots of these trees are hashed together to form a FORS root.  SLH-DSA uses a one-time signature scheme called WOTS+. The FORS tree roots are signed by a WOTS+ one-time signature private key.  The corresponding WOTS+ public keys form the leaves in d-layers of Merkle subtrees in the SLH-DSA hypertree.  The bottom layer of that hypertree signs the FORS roots with WOTS+. The root of the bottom Merkle subtrees are then signed with WOTS+ and the corresponding WOTS+ public keys form the leaves of the next level up subtree.  Subtree roots are consequently signed by their corresponding subtree layers until we reach the top subtree.  The top layer subtree forms the hypertree root which is trusted at the verifier.
+
+A SLH-DSA signature consists of the FORS signature, the WOTS+ signature in each layer, and the path to the root of each subtree until the root of the hypertree is reached.
```
Comment (BashiriK) [inline draft-gazdag-x509-slhdsa.md:107]: Though of negligible size, but there is also a randomizer included in the signature.

File: draft-gazdag-x509-slhdsa.md [hunk 2]
```diff
@@ -43,40 +54,434 @@ author:
...
+
+SLH-DSA is a stateless hash-based signature algorithm.  Stateful hash-based signature schemes require that the WOTS+ private key (generated by using a state index) is never reused or the scheme loses it security.  Although its security decreases, FORS which is used at the bottom of the SLH-DSA hypertree does not collapse if the same private key used to sign two or more different messages like in stateful hash-based signature schemes.  Without the need for state kept by the signer to ensure it is not reused, SLH-DSA is much less fragile.
+
+SLH-DSA was designed to sign up to 2^64 messages and offers three security levels.  The parameters of the SLH-DSA hypertree include the security parameter, the hash function, the tree height, the number of layers of subtrees, the Winternitz parameter of WOTS+, the number of FORS trees and leaves in each.  The parameters for each of the security levels were chosen to provide 128 bits of security, 192 bits of security, and 256 bits of security.
```
Comment (BashiriK) [inline draft-gazdag-x509-slhdsa.md:113]: Above the hypertree part of SLH-DSA was separated from the FORS part. However, here it is written that "The parameters of the SLH-DSA hypertree include [...] the number of FORS trees". I would change it to 
"The parameters of SLH-DSA include ...".

Comment (BashiriK) [reply draft-gazdag-x509-slhdsa.md:113]: My opinion: If we include the hash function as a paramter of the scheme, we should also include "small vs. fast" as a parameter.

File: draft-gazdag-x509-slhdsa.md [hunk 3]
```diff
@@ -43,40 +54,434 @@ author:
...
+    |  [RFC5912] for ASN.1 copmatible with the 2015 ASN.1 syntax.
+~~~
+
+The fields in AlgorithmIdentifier have the following meanings:
+
+* algorithm identifies the cryptographic algorithm with an object identifier.
+
+* parameters, which are optional, are the associated parameters for the algorithm identifier in the algorithm field.
+
+The OIDs are:
```
Comment (BashiriK) [inline draft-gazdag-x509-slhdsa.md:139]: Did we decide to leave out SHA2-parameters for level III and V due to the discussion on the forum? Is it really decided in that way? If yes, I missed that.

File: draft-gazdag-x509-slhdsa.md [hunk 4]
```diff
@@ -43,40 +54,434 @@ author:
...
+    id-alg-slh-dsa-256f-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+~~~
+
+The contents of the parameters component for each algorithm are absent.
+
+# SLH-DSA Signatures in PKIX
+
+SLH-DSA is a digital signature scheme built upon hash functions. The security of SLH-DSA relies on the presumed diffculty of finding preimages for hash functions as well as several related properties of the same hash functions.
```
Comment (BashiriK) [inline draft-gazdag-x509-slhdsa.md:179]: My optinion: This paragraph belongs to the section "SLH-DSA Hash-based Signature Algorithm Overview"

**Review by fluppe2 — APPROVED** [review]

**Pull request merged**